crates/containerd-shim-wasm/src/sandbox/shim/otel.rs

```diff
@@ -134,31 +151,208 @@ impl Drop for ShutdownGuard {
 
 #[derive(Default)]
 pub struct ConfigBuilder {
-    otel_endpoint: Option<String>,
-    otel_protocol: Option<String>,
+    traces_endpoint: Option<String>,
+    traces_protocol: String,
 }
 
 impl ConfigBuilder {
-    /// Sets the OpenTelemetry endpoint.
-    pub fn otel_endpoint(mut self, otel_endpoint: String) -> Self {
-        self.otel_endpoint = Some(otel_endpoint);
+    /// Sets the OTLP endpoint from environment variables.
+    pub fn traces_endpoint_from_env(mut self) -> Self {
+        self.traces_endpoint = env::var(OTEL_EXPORTER_OTLP_TRACES_ENDPOINT)
+            .or_else(|_| env::var(OTEL_EXPORTER_OTLP_ENDPOINT))
+            .ok();
         self
     }
 
-    pub fn otel_protocol(mut self, otel_protocol: String) -> Self {
-        self.otel_protocol = Some(otel_protocol);
+    /// Sets the OTLP protocol from environment variables.
+    pub fn traces_protocol_from_env(mut self) -> Self {
+        self.traces_protocol = env::var(OTEL_EXPORTER_OTLP_TRACES_PROTOCOL).unwrap_or(
+            env::var(OTEL_EXPORTER_OTLP_PROTOCOL)
+                .unwrap_or(OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT.to_owned()),
+        );
         self
     }
 
     /// Builds the `OtelConfig` instance.
     pub fn build(self) -> Result<Config, &'static str> {
-        let otel_endpoint = self.otel_endpoint.ok_or("otel_endpoint is required")?;
-        let otel_protocol = self
-            .otel_protocol
-            .unwrap_or_else(|| OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT.to_owned());
         Ok(Config {
-            otel_endpoint,
-            otel_protocol,
+            traces_endpoint: self.traces_endpoint.ok_or("otel_endpoint is required")?,
+            traces_protocol: self.traces_protocol,
```

Comment:
**question:** protocol is also required, otherwise it ends hitting the error in L.128, right?